pkg/controller/machinepool/azureactuator.go

```diff
@@ -95,6 +98,24 @@ func (a *AzureActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *
 		workerRole,
 		workerUserDataName,
 	)
+
+	// Use image set for machinepool azure platform instead of the default
+	if image := pool.Spec.Platform.Azure.Image; (machineapi.Image{}) != image {
+		for _, ms := range installerMachineSets {
+			provider := ms.Spec.Template.Spec.ProviderSpec.Value.Object.(*azureprovider.AzureMachineProviderSpec)
+			provider.Image = azureprovider.Image{
+				Publisher:  image.Publisher,
+				Offer:      image.Offer,
+				SKU:        image.SKU,
+				Version:    image.Version,
+				ResourceID: image.ResourceID,
```

Comment:
Noting that (OpenShift Machine API) `Image.Type` is not passed on to the Azure Provider Image. These are the only fields supported by the Azure Provider.

Reply:
These types are moving into the api repo. I'm going to revendor 4.11 to pick up those changes which include `Image.Type`.